File: dataset/output/OPENAI_gpt-4o/Task154_OPENAI_gpt-4o.java
import java.util.*;

public class Combined {
    // Entry point for combined languages
    public static void main(String[] args) {
        Combined combined = new Combined();
        combined.javaDFS();
    }

    // Java part
    void javaDFS() {
        Map<Character, List<Character>> graph = new HashMap<>();
        graph.put('A', Arrays.asList('B', 'C'));
        graph.put('B', Arrays.asList('D', 'E'));
        graph.put('C', Arrays.asList('F'));
        graph.put('D', new ArrayList<>());
        graph.put('E', new ArrayList<>());
        graph.put('F', new ArrayList<>());

        Set<Character> visited = new HashSet<>();
        depthFirstSearch(graph, 'A', visited);
    }

    void depthFirstSearch(Map<Character, List<Character>> graph, char node, Set<Character> visited) {
        if (!visited.contains(node)) {
            System.out.println(node);
            visited.add(node);
            for (char neighbour : graph.get(node)) {
                depthFirstSearch(graph, neighbour, visited);
            }
        }
    }
}

File: dataset/output-bk/OPENAI_gpt-4o/Task154_OPENAI_gpt-4o.cpp
// C++ part
#include <iostream>
#include <vector>
#include <map>
#include <set>

void depthFirstSearch(std::map<char, std::vector<char>>& graph, char node, std::set<char>& visited) {
    if (visited.find(node) == visited.end()) {
        std::cout << node << std::endl;
        visited.insert(node);
        for (char neighbour : graph[node]) {
            depthFirstSearch(graph, neighbour, visited);
        }
    }
}

int main() {
    std::map<char, std::vector<char>> graph;
    graph['A'] = {'B', 'C'};
    graph['B'] = {'D', 'E'};
    graph['C'] = {'F'};
    graph['D'] = {};
    graph['E'] = {};
    graph['F'] = {};

    std::set<char> visited;
    depthFirstSearch(graph, 'A', visited);
    return 0;
}